### PR TITLE
Remove storage service from supervisor

### PR DESCRIPTION
**Pull request by vcfgv**
## What do these changes do?

1. Remove storage service from supervisor. Now storage service on supervisor will remain empty.
2. Under mars backend, large UDF will be serialized and saved as op.func while tiling and will be deserialized while executing.
3. Under ray backend (oscar and dag), large UDF will be put into ObjectStore while tiling and brought back while executing.

## Related issue number

#3193 

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/dataframe/utils.py [hunk 1]
```diff
@@ -1464,21 +1455,44 @@ def clean_up_func(op):
...
+        if _is_on_ray(ctx):
+            import ray
+
+            op.func_key = ray.put(op.func)
+            op.func = None
+        else:
+            op.func = cloudpickle.dumps(op.func)
+
+
+def _is_on_ray(ctx):
```
Comment (hekaisheng) [inline mars/dataframe/utils.py:1473]: Better to move this util to top level.

File: mars/dataframe/utils.py [hunk 2]
```diff
@@ -1464,21 +1455,44 @@ def clean_up_func(op):
...
+    #   while get_worker_addresses method isn't currently implemented in RayExecutionWorkerContext.
+    try:
+        worker_addresses = ctx.get_worker_addresses()
+    except AttributeError:  # pragma: no cover
+        assert isinstance(ctx, RayExecutionWorkerContext)
+        return True
+    return isinstance(ctx, RayExecutionContext) or is_ray_address(worker_addresses[0])
 
 
 def restore_func(ctx: Context, op):
```
Comment (hekaisheng) [inline mars/dataframe/utils.py:1490]: As discussed earlier, add a TODO about removing this function in the future. Construct a wrapped class for functions, override `__getstate__` and `__setstate__` to address these serialization issues.

**Review by hekaisheng — COMMENTED** [review]
I left two comments, this PR is acceptable overall for now, looking forward the refactor in following PRs.

**Review by chaokunyang — DISMISSED** [review]
LGTM

**Review by aresnow1 — APPROVED** [review]

**Review by hekaisheng — APPROVED** [review]
LGTM

**Review by chaokunyang — APPROVED** [review]
LGTM

**Pull request merged**